single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/styles/iidm/OverloadHighlightStyleProvider.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.sld.svg.styles.iidm;
+
+import com.powsybl.iidm.network.*;
+import com.powsybl.sld.library.ComponentLibrary;
+import com.powsybl.sld.model.graphs.Graph;
+import com.powsybl.sld.model.graphs.VoltageLevelGraph;
+import com.powsybl.sld.model.nodes.*;
+import com.powsybl.sld.model.nodes.feeders.FeederWithSides;
+import com.powsybl.sld.svg.styles.EmptyStyleProvider;
+import java.util.*;
+
+import static com.powsybl.sld.svg.styles.StyleClassConstants.*;
+
+/**
+ * @author Jamal KHEYYAD {@literal <jamal.kheyyad at rte-international.com>}
+ */
+public class OverloadHighlightStyleProvider extends EmptyStyleProvider {
+    private final Network network;
+
+    public OverloadHighlightStyleProvider(Network network) {
+        this.network = network;
+    }
+
+    @Override
+    public List<String> getEdgeStyles(Graph graph, Edge edge) {
+        Optional<String> overloadStyle = getOverloadStyle(edge);
+        return overloadStyle.map(Collections::singletonList).orElse(Collections.emptyList());
+    }
+
+    private Optional<String> getOverloadStyle(Edge edge) {
+        List<Node> nodes = edge.getNodes();
+        for (Node node : nodes) {
+            if (node instanceof FeederNode feederNode) {
+                return isOverloaded(feederNode) ? Optional.of(OVERLOAD_STYLE_CLASS) : Optional.empty();
+            }
+        }
+        return Optional.empty();
+    }
+
+    private boolean isOverloaded(FeederNode n) {
+        if (!(n.getFeeder() instanceof FeederWithSides)) {
+            return false;
+        }
+        Branch<?> branch = network.getBranch(n.getEquipmentId());
+        if (branch != null && branch.isOverloaded()) {
+            return true;
```

Comment:
if the branch is not null and not overloaded you don't want to look for a three-winding transformer, so you'd rather write
```suggestion
        if (branch != null) {
            return branch.isOverloaded();
```

Reply:
done